src/jit/emitarm64.cpp

```diff
@@ -7864,6 +7940,78 @@ BYTE*               emitter::emitOutputLJ(insGroup  *ig, BYTE *dst, instrDesc *i
     srcOffs = emitCurCodeOffs(dst);
     srcAddr = emitOffsetToPtr(srcOffs);
 
+    if (loadConstant)
+    {
+        /* This is actually a reference to the JIT data section */
+        int doff = id->idAddr()->iiaGetJitDataOffset();
+        assert(doff >= 0);
+        ssize_t imm = emitGetInsSC(id);
+        assert((imm >= 0) && (imm < 0x1000));   // 0x1000 is arbitrary, currently 'imm' is always 0
+
+        unsigned dataOffs = (unsigned)(doff + imm);
+        assert(dataOffs < emitDataSize());
+        dstAddr = emitDataOffsetToPtr(dataOffs);
+
+        regNumber dstReg = id->idReg1();
+        regNumber addrReg = dstReg; // an integer register to compute long address.
+        emitAttr opSize = id->idOpSize();
+
+        if (id->idjShort)
+        {
+            // ldr x/v, [rel addr] -- load constant from current addr(ip) + rel addr.
+            assert(ins == INS_ldr);
+            assert(fmt == IF_LS_1A);
+            distVal = (ssize_t)(dstAddr - srcAddr);
+            dst = emitOutputShortConstant(dst, ins, fmt, distVal, dstReg, opSize);
+        }
+        else
+        {
+            // adrp x, [rel page addr] -- compute page address: current page addr + rel page addr
+            assert(fmt == IF_LARGELDC);
+            ssize_t relPageAddr = (((ssize_t)dstAddr & 0xFFFFFFFFFFFFF000LL) - ((ssize_t)srcAddr & 0xFFFFFFFFFFFFF000LL)) >> 12;
+            if (isVectorRegister(dstReg))
+            {
+                // Update addrReg with the reserved integer register
+                // since we cannot use dstReg (vector) to load constant directly from memory.
+                addrReg = id->idReg2();
+                assert(isGeneralRegister(addrReg));
+            }
+            ins = INS_adrp;
+            fmt = IF_DI_1E;
+            dst = emitOutputShortAddress(dst, ins, fmt, relPageAddr, addrReg);
+
+            // ldr x, [x, page offs] -- load constant from page address + page offset into integer register.
+            ssize_t imm12 = (ssize_t)dstAddr & 0xFFF; // 12 bits
+            assert(isValidUimm12(imm12));
+            ins = INS_ldr;
+            fmt = IF_LS_2B;
+            dst = emitOutputShortConstant(dst, ins, fmt, imm12, addrReg, opSize);
+
+            // fmov v, d -- copy constant in integer register to vector register.
```

Comment:
Instead of loading the floating point constant into the integer registers above.
why don't we change the ldr instruction to an add with 12-byte immediate
thus computing the proper address to load from
then do a floating point load instruction with a zero offset